packages/babel-parser/src/parser/comments.js

```diff
@@ -84,85 +126,20 @@ export default class CommentsParser extends BaseParser {
-      const lastComment = last(this.state.leadingComments);
-
-      if (firstChild.type === "ObjectProperty") {
-        if (lastComment.start >= node.start && lastComment.end <= node.end) {
-          if (this.state.commentPreviousNode) {
-            for (j = 0; j < this.state.leadingComments.length; j++) {
-              if (
-                this.state.leadingComments[j].end <
-                this.state.commentPreviousNode.end
-              ) {
-                this.state.leadingComments.splice(j, 1);
-                j--;
-              }
-            }
-
-            if (this.state.leadingComments.length > 0) {
-              firstChild.trailingComments = this.state.leadingComments;
-              this.state.leadingComments = [];
-            }
-          }
-        }
-      } else if (
-        node.type === "CallExpression" &&
-        node.arguments &&
-        node.arguments.length
-      ) {
-        const lastArg = last(node.arguments);
-
-        if (
-          lastArg &&
-          lastComment.start >= lastArg.start &&
-          lastComment.end <= node.end
-        ) {
-          if (this.state.commentPreviousNode) {
-            for (j = 0; j < this.state.leadingComments.length; j++) {
-              if (
-                this.state.leadingComments[j].end <
-                this.state.commentPreviousNode.end
-              ) {
-                this.state.leadingComments.splice(j, 1);
-                j--;
-              }
-            }
-            if (this.state.leadingComments.length > 0) {
-              lastArg.trailingComments = this.state.leadingComments;
-              this.state.leadingComments = [];
-            }
-          }
-        }
-      } else if (node.type === "ArrayExpression" && node.elements.length > 0) {
-        if (this.state.commentPreviousNode) {
-          for (j = 0; j < this.state.leadingComments.length; j++) {
-            if (
-              this.state.leadingComments[j].end <
-              this.state.commentPreviousNode.end
-            ) {
-              this.state.leadingComments.splice(j, 1);
-              j--;
-            }
-          }
-
-          const lastElement = last(node.elements);
-          lastElement.trailingComments = [];
-          while (this.state.leadingComments.length) {
-            const leadingComment = this.state.leadingComments.shift();
-            if (leadingComment.end < node.end) {
-              lastElement.trailingComments.push(leadingComment);
-            } else {
-              if (node.trailingComments === undefined) {
-                node.trailingComments = [];
-              }
-              node.trailingComments.push(leadingComment);
-            }
-          }
-        }
+    // Adjust comments that follow a trailing comma on the last element in a
+    // comma separated list of nodes to be the trailing comments on the last
+    // element
+    if (firstChild) {
+      switch (node.type) {
```

Comment:
Maybe also `SequenceExpression`? (I didn't check)

Reply:
I don't think `SequenceExpression` will allow trailing comma.

Reply:
I skimmed through https://github.com/babel/babylon/blob/master/ast/spec.md and looks like `ImportDeclaration` and `ArrayPattern` also allow trailing commas. Should I add those?

Reply:
Yeas please! And also `ExportNamedDeclaration` and `ObjectPattern`